cdap-common/src/main/java/io/cdap/cdap/common/metadata/QueryParser.java

```diff
@@ -58,10 +71,69 @@ private QueryParser() {}
     return queryTerms;
   }
 
+
   private static QueryTerm parseQueryTerm(String term) {
-    if (term.charAt(0) == REQUIRED_OPERATOR && term.length() > 1) {
-      return new QueryTerm(term.substring(1), Qualifier.REQUIRED);
+    String endTerm;
+    Qualifier endQualifier;
+    SearchType endSearchType;
+    Comparison endComparison;
```

Comment:
Should be able to declare these four variables and initialize them with the ternary operation in the same line.

Reply:
Good point. Addressed in latest commit.